src/main/java/com/github/bordertech/config/DefaultConfiguration.java

```diff
@@ -1499,21 +1316,21 @@ protected String get(final String key, final String defolt) {
 	 */
 	protected String get(final String key) {
 		// Check environment property
-		if (useEnvironmentKey(key)) {
-			String result = backing.get(getEnvironmentKey(key));
+		if (useProfileKey(key)) {
+			String result = backing.get(getProfileKey(key));
 			if (result != null) {
-				return result;
+				return StringSubstitutor.replace(result, backing);
 			}
 		}
-		return backing.get(key);
+		//Final substitution check
+		return StringSubstitutor.replace(backing.get(key), backing);
```

Comment:
Just want to make sure I understand. Why do we need the `StringSubstititor` in this `get` method if we have already called `substitute` after the properties were loaded.

Reply:
As discussed, I was catering for property definitions defined in the incorrect order. i.e. User error rather than system error.

I have refactored to not substitute on every get() call but to only substitute on the existing load  and now have added when the addOrModifyProperty() is called to be able to substitute at runtime.